### PR TITLE
Wagtail 4.2 compatibility

### PR DESCRIPTION
**Pull request by nickmoreton**
This updates the package to work with Wagtail 4.2

and drops support for wagtail versions < 4.1

Support ticket: https://torchbox.monday.com/boards/1124794299/pulses/1131929727

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -21,7 +21,8 @@ deps =
     djangomain: git+https://github.com/django/django.git@main#egg=Django
     wagtail2.15: wagtail>=2.15,<2.16
     wagtail3: wagtail>=3.0,<4.0
-    wagtail4: wagtail>=4.0,<5.0
+    wagtail4: wagtail>=4.0,<4.2
+    wagtail42: wagtail==4.2rc1
```
Comment (zerolab) [inline tox.ini:25]:
```suggestion
    wagtail42: wagtail>=4.2rc1,<5.0
```

File: src/wagtailmarkdown/edit_handlers.py [hunk 2]
```diff
@@ -1,6 +1,12 @@
 import warnings
 
-from wagtail.admin.edit_handlers import FieldPanel
+from wagtail import VERSION as WAGTAIL_VERSION
+
+
+if WAGTAIL_VERSION >= (3, 0):
+    from wagtail.admin.panels import FieldPanel
+else:
+    from wagtail.admin.edit_handlers import FieldPanel
```
Comment (zerolab) [inline src/wagtailmarkdown/edit_handlers.py:9]: we might as well drop support for Wagtail < 4.1

Comment (nickmoreton) [reply src/wagtailmarkdown/edit_handlers.py:9]: We were talking about this today in our meeting. Do you mean we should remove these conditionals and also no longer test versions <4.1 in tox and/or CI

It's probably a more general question as well, aimed at the other packages we'll be updating in the coming weeks 😄 
Thanks

Comment (zerolab) [reply src/wagtailmarkdown/edit_handlers.py:9]: Now that 2.15 is no longer supported (https://endoflife.date/wagtail), my preference is to drop pre 4.1 logic and support. So this means dropping conditionals, tests on CI etc. I would wager we should even make the lower Wagtail version 4.1 too


Now, this is my preference on the packages I maintain. Different packages will have different preferences.

Comment (zerolab) [reply src/wagtailmarkdown/edit_handlers.py:9]: Also, https://github.com/wagtail/wagtail/wiki/Python-Package-Maintenance-Guidelines#setuppy--setupcfg--pyprojecttoml

> Support for EOL versions of Python, Django and Wagtail should be removed from the earliest minor release of the package following the moment the dependency went EOL. This does not apply if provision is made to maintain support for the EOL dependency.

Comment (nickmoreton) [reply src/wagtailmarkdown/edit_handlers.py:9]: Excellent ^^ thank you.

File: tox.ini (hunk 1 above)
Comment (nickmoreton) [reply tox.ini]: This has now been changed

File: tox.ini [hunk 3]
```diff
@@ -21,7 +21,8 @@ deps =
     djangomain: git+https://github.com/django/django.git@main#egg=Django
     wagtail2.15: wagtail>=2.15,<2.16
     wagtail3: wagtail>=3.0,<4.0
-    wagtail4: wagtail>=4.0,<5.0
+    wagtail4: wagtail>=4.0,<4.2
```
[the remark this answers is not in the document]
Comment (nickmoreton) [reply tox.ini]: This has now been changed

File: src/wagtailmarkdown/edit_handlers.py (hunk 2 above)
Comment (nickmoreton) [reply src/wagtailmarkdown/edit_handlers.py]: I have removed all Wagtail v3 version checks

**Pull request merged**